src/dune_rules/artifact_substitution.ml

```diff
@@ -94,11 +94,14 @@ module Conf = struct
   ;;
 
   let sign_hook_of_context (context : Context.t) =
-    let config = context.ocaml.ocaml_config in
+    let config = (Context.ocaml context).ocaml_config in
     match Ocaml_config.system config, Ocaml_config.architecture config with
     | "macosx", "arm64" ->
       let codesign_name = "codesign" in
-      (match Bin.which ~path:context.path codesign_name with
+      (match
+         let path = Context.path context in
```

Comment:
Why not just inline this?

Reply:
No reason. It's a huge PR and I don't have time to optimize every little instance.